packages/protoplugin-example/protoc-gen-twirp-es/src/typescript.ts

```diff
@@ -0,0 +1,110 @@
+// Copyright 2021-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { MethodKind } from "@bufbuild/protobuf";
+import type { Schema } from "@bufbuild/protoplugin/ecmascript";
+import {
+  literalString,
+  makeJsDoc,
+  localName,
+} from "@bufbuild/protoplugin/ecmascript";
+
+// prettier-ignore
+export function generateTs(schema: Schema) {
+  for (const file of schema.files) {
+    const f = schema.generateFile(file.name + "_twirp.ts");
+    f.preamble(file);
+    f.print("import type { JsonValue, Message } from '@bufbuild/protobuf';");
+    f.print();
+    f.print("export interface TransportOptions {");
+    f.print("    baseUrl: string;");
+    f.print("    headers?: HeadersInit;");
+    f.print("}");
+    f.print();
+    f.print("class TwirpClient {");
```

Comment:
With `TransportOptions`, this looks a bit close to connect-web transports. It's probably feasible to implement Twirp as a connect-web transport, but are we going this route here?

I'd say we don't need `class TwirpClient` here, and could simply generate the entire fetch() call in the method. We don't really need `createElizaServiceClient()` either, because it's already a class that can be used with `new`. The generated `interface ElizaService` would be useful for mocking in tests, but for the sake of a simple example, does it add any feature we'd like to show that we don't already show with the generated class?